figures/scripts/fig3-panel-tp53.R

```diff
@@ -0,0 +1,182 @@
+  data_dir, 
+  "pbta-snv-consensus-mutation-tmb-coding.tsv"
+)
+tmb_coding_df <- read_tsv(tmb_coding_file)
+
+
+# Rename DNA column for joining purposes and select columns of interest
+tp53_scores_df_subset <- tp53_scores_df %>%
+  rename(Kids_First_Biospecimen_ID = Kids_First_Biospecimen_ID_DNA) %>%
+  select(Kids_First_Biospecimen_ID, 
+         tp53_score) %>%
+  distinct() %>%
+  # remove all NAs, which exist in IDs and tp53
+  drop_na()
+  
+# Get coding samples of interest and annotate samples as normal, hyper, or ultra
+mutators <- tmb_coding_df %>%
+  # columns of interest
+  select(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode,
+         tmb) %>%
+  # keep only rows of interest
+  filter(Kids_First_Biospecimen_ID %in% tp53_scores_df_subset$Kids_First_Biospecimen_ID) %>%
+  # add a column about mutator
+  mutate(
+    mutator = case_when(
+      tmb < 10 ~ "Normal",
+      tmb >= 10 & tmb < 100 ~ "Hypermutant",
+      tmb >= 100 ~ "Ultrahypermutant")
+  )
+
+# Join data for visualization
+tp53_plot_df <- tp53_scores_df_subset %>%
+  inner_join(mutators, by="Kids_First_Biospecimen_ID") %>%
+  inner_join(
+    select(histologies_df, 
+           Kids_First_Biospecimen_ID,
+           cancer_group)
+  ) %>%
+  # filter to cancer groups and wrap for x-axis
+  filter(cancer_group %in% cancer_groups_to_plot) %>%
+  mutate(cancer_group = str_wrap(cancer_group, 20))
+
+
+#### Plot and associated legend --------------------------------------------------
+
+
+# Define colors to use
+legend_colors <- c(Normal      = "grey40",
+                   Hypermutant = "orange",
+                   Ultrahypermutant = "red")
+
+# Boxplot with overlayed jitter with colors "mapped" to `mutator`
+set.seed(9) # reproducible jitter to ensure we can see all N=6 points
+tp53_tmb_boxplot <- ggplot(tp53_plot_df) +
+  aes(
+    x = fct_reorder(cancer_group, tp53_score), # order cancer groups by tp53 score
+    y = tp53_score
+  ) +
+  geom_boxplot(
+    outlier.shape = NA, # no outliers
+    color = "grey20",    # dark grey color
+    size = 0.4 
+  ) +
+  # Separate out jitters so that the mutant layers are ON TOP OF normal
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Normal",],
+    width = 0.15, 
+    alpha = 0.7,
+    color = legend_colors["Normal"]
+  ) +
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Hypermutant",],
+    width = 0.15, 
+    pch = 21,
+    size = 2.25,
+    fill = legend_colors["Hypermutant"]
+  ) +  
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Ultrahypermutant",],
+    width = 0.15, 
+    pch = 21,
+    size = 2.25,
+    fill = legend_colors["Ultrahypermutant"]
+  ) +  
+  labs(x = "Cancer group", 
+       y = "TP53 Score") +
+  # do we want an hline at 0.5? Might be useful guiding line but also add unnecessary visual noise
+  # geom_hline(yintercept = 0.5) +
+  ggpubr::theme_pubr() +
+  theme(
+    axis.text.x = element_text(angle = 45, hjust=1, size = rel(0.8))
+  )
```

Comment:
I'm a little confused about why we see `grey40` outlines of the normal points and `black` outlines of the mutator points here given that they're all using `geom_jitter()` with the fill specified, but not the color?

Reply:
What is _expected_ are regular points (just a color, no fill) for normal, but then black-outlined with fills for mutators. I found that this divide makes the mutators stand out much more strongly. 

The grey outlines are a weird graphics situation that occurs with `alpha`, and why exactly this happens I don't know. I can try adding explicit shape args like  `pch=19` or `pch=20` into the "normal" jitter to see if we can get rid of it that outline.

Reply:
I think it's fine so long as the legend will be clear.